src/config/reader.rs

```diff
@@ -116,20 +116,32 @@ impl ConfigReader {
                     }
                 }
                 LinkType::Protobuf => {
-                    let mut descriptors = self
-                        .resolve_descriptors(self.read_proto(&source.path).await?)
-                        .await?;
+                    let parent_descriptor = self.read_proto(&source.path).await?;
+                    let id = parent_descriptor
+                        .package // we should ideally take the package name as default id
+                        .clone()
+                        .unwrap_or(
+                            config_link
+                                .id
+                                .to_owned() // if there is no package id then we use link id
+                                .unwrap_or(
+                                    parent_descriptor
+                                        .name // if none of them is available,
+                                        // then we use file name.
+                                        .clone()
+                                        .unwrap_or_default(),
```

Comment:
I don't understand how it will work from user perspective.
if we have package inside proto file it seems ok, but in other cases I need really think link ids or filenames and they overrides each other. 
Also, the `name` is the path to file, not the file name. And we may end up with empty string here that is even more unclear how it should work.

Reply:
`name` must be proto name, we have a faulty implementation of `protox_parse` as I told on discord.
https://github.com/tailcallhq/tailcall/blob/f4306dfcd629d0b2e477dda2ce4096ba18083965/src/config/reader.rs#L263

it just puts the path as name, but the path might be a url or full path, we need some way to avoid that. I think we can have 3 cases, if the proto file have package, then we just use it everywhere, else we check if the path ends with `*.proto`, else we throw an exception about naming or package id

Reply:
when trying to fix the original issue with package name I faced the issue described in the [comment](https://github.com/tailcallhq/tailcall/pull/1198/files#diff-292fd7f2de6455aa7a8afbf0b7a1d0f4a7538f4f1c92ba3be3e29b914d749867R142).

I think the most easiest way for now is to just require package in proto and throw meaningful validation error otherwise. That will clear and obvious way to work with without any mentioned gotchas for backend implementation. And depending on future use we'll think if that require changes cuz proto files without package could be not very used for cases except simple testing